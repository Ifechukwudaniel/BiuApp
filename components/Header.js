import React,{useState, useEffect, useRef, useImperativeHandle, forwardRef} from 'react';
import {  Header, Left, Body, Right, Button, Title, Text, Icon,  } from 'native-base'
import {Image, StatusBar, Dimensions, Animated, Platform, Easing, SafeAreaView, View} from 'react-native'
import EStyleSheet from 'react-native-extended-stylesheet'
import HandBugger from '../assets/svg/handBugger.svg' 
const entireScreenWidth = Dimensions.get('window').width;
const rem = entireScreenWidth/380
import Constants from 'expo-constants';
import { TouchableOpacity } from 'react-native-gesture-handler';
const statusBarHeight = Constants.statusBarHeight


const HeaderComponent = forwardRef((props, ref)=> {
    const [animation, setAnimation] = useState(false)
    const [value] = useState(new Animated.Value(0))
   
    useEffect(()=>{
      Animated.timing(value, {
          toValue: animation? 1 : 0,
           easing:Easing.bounce,
          duration: Platform.select({
            ios:150,
            android:50 
          }),
          useNativeDriver:true
      }).start()
  }, [animation])
   
  const height = value.interpolate({
    inputRange: [0, 1],
    outputRange: [0*rem, -110*rem],  
  })
   
  useImperativeHandle(ref, () => {
     return {
      animateUp(){
        setAnimation(true)  
      },
      animateDown(){
        setAnimation(false)  
      }
     }
  });
   
  if (Platform.OS=="android") {
    return (
      <Animated.View style={Platform.select({
        android:{
          marginTop:statusBarHeight
        },
        ios:{
          marginTop:0
        }
      }),{
        transform:[
          {
            translateY:height,
          }
        ]
      }}>
      <View transparent  style={[styles.header]}>
       <StatusBar hidden={false} backgroundColor="#0C0C0E" barStyle="light-content" />
         <View style={styles.Left}>
           <Button style={styles.HandBugger} onPress={()=>props.navigation.openDrawer()} transparent>
             <HandBugger width={30*rem} height={30*rem}/>
           </Button>
         </View>
         <View  style={styles.center}>
           <Title style={styles.title}>{props.screenName}</Title>
         </View>
         <View style={styles.right}>
           { !props.back ?   
           <Button   onPress={()=>{
           props.navigation.navigate('Settings')
           }}   transparent>
             <Image resizeMode="cover" style={styles.userImage} source={require('../assets/images/image.jpeg')}/>
           </Button>
           : 
           <Button  onPress={()=>props.navigation.navigate(props.backScreen ? props.backScreen :'Home' )}   transparent>
              <Icon style={{color:'#FF912C'}} type="FontAwesome5"  name="arrow-left"/>
           </Button>
           }
         </View>
      </View>
      </Animated.View>
   );
  }
  else{
    return (
      <Animated.View style={{
        transform:[
          {
            translateY:height
          }
        ]
      }}>
      <Header transparent  style={[styles.header]}>
       <StatusBar backgroundColor="transparent" hidden={false} barStyle="light-content"/>
         <Left style={{flex:1}}>
           <Button onPress={()=>props.navigation.openDrawer()} transparent>
             <HandBugger width={30*rem} height={30*rem}/>
           </Button>
         </Left>
         <Body  style={{flex:1}}>
           <Title style={styles.textStyle}>{props.screenName}</Title>
         </Body>
         <Right style={{flex:1}}>
           { !props.back ?   
           <Button  onPress={()=>{
         //    setAnimation(true)
           props.navigation.navigate('Settings')
           }}   transparent>
             <Image resizeMode="cover" style={styles.userImage} source={require('../assets/images/image.jpeg')}/>
           </Button>
           : 
           <Button  onPress={()=>props.navigation.navigate(props.backScreen ? props.backScreen :'Home' )}   transparent>
              <Icon style={{color:'#FF912C'}} type="FontAwesome5"  name="arrow-left"/>
           </Button>
           }
         </Right>
      </Header>
      </Animated.View>
   );
  }
   
})


const styles = EStyleSheet.create(
   Platform.select({
     android:{
      userImage:{
        height: '45rem',
        width:'45rem',
        borderRadius: '45rem',
        borderWidth: '2rem',
        borderColor: "#FF912C",
        alignSelf: 'center',
        marginTop: '2rem',
    },
    textStyle:{
        color:"#fff",
        alignSelf: 'center',
        paddingTop:0,
        justifyContent: 'center',
    },
    header:{
      flexDirection: 'row',
      width:'100%',
      backgroundColor: "#0C0C0E",
      height:"65rem",
      borderWidth:'0rem',
      borderColor:'#0c0c0e',
      alignSelf: 'center',
    },
    left:{
      width:'15%',
    },
    HandBugger:{
      marginLeft:'20rem',
      marginTop: '5rem',
    },
    title:{
      marginTop: '17rem',
      color:"#fff",
      alignSelf: 'center',
      justifyContent: 'center',
     },
    right:{
      width:'15%',
      marginTop:'6rem',
      marginLeft:'5rem'
    },
    center:{
      width:'70%',
      marginLeft:'5rem'
    }
    },
    ios:{
      header:{
        backgroundColor: "#0C0C0E",
        paddingTop: '48rem',
        paddingBottom: '20rem',
        paddingLeft: '20rem',
        height:"90rem",
        borderWidth:'0rem',
        borderColor:'#0c0c0e',
     },
     userImage:{
         height: '50rem',
         width:'50rem',
         borderRadius: '70rem',
         borderWidth: '2rem',
         borderColor: "#FF912C",
         alignSelf: 'center',
     },
     textStyle:{
         color:"#fff",
         alignSelf: 'center',
         paddingTop:0,
         justifyContent: 'center',
     }
    }
   })
)
export default HeaderComponent;