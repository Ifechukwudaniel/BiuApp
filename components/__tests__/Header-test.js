import React from 'react';
import renderer from 'react-test-renderer';

import Header from '../Header';
import HeaderComponent from '../Header';

it(`Header renders correctly`, () => {
  const tree = renderer.create(<HeaderComponent screenNAme="TEST"/>).toJSON();

  expect(tree).toMatchSnapshot();
});
