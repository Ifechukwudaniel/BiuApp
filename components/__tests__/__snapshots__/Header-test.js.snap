// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<View>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "#F8F8F8",
            "borderBottomColor": "#a7a6ab",
            "borderBottomWidth": 0.5,
            "elevation": 3,
            "flexDirection": "row",
            "height": 64,
            "justifyContent": "center",
            "left": 0,
            "paddingLeft": 6,
            "paddingRight": 10,
            "paddingTop": 18,
            "right": 0,
            "shadowColor": undefined,
            "shadowOffset": undefined,
            "shadowOpacity": undefined,
            "shadowRadius": undefined,
            "top": 0,
          },
          undefined,
          undefined,
        ]
      }
    >
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "alignSelf": null,
              "backgroundColor": "transparent",
              "borderBottomWidth": null,
              "borderColor": null,
              "borderLeftWidth": null,
              "borderRadius": 5,
              "borderRightWidth": null,
              "borderTopWidth": null,
              "borderWidth": undefined,
              "elevation": 0,
              "flexDirection": "row",
              "height": 45,
              "justifyContent": "space-between",
              "marginLeft": -3,
              "opacity": 1,
              "paddingBottom": 6,
              "paddingLeft": 4,
              "paddingRight": 6,
              "paddingTop": 6,
              "shadowColor": null,
              "shadowOffset": null,
              "shadowOpacity": null,
              "shadowRadius": null,
            }
          }
        >
          <Text />
          <Text
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#007aff",
                "fontFamily": "System",
                "fontSize": 17,
                "marginLeft": 0,
                "marginRight": 0,
                "paddingLeft": 2,
                "paddingRight": undefined,
                "top": 1,
              }
            }
            uppercase={false}
          >
            Back
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "flex": 1,
          }
        }
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "color": "#000",
                "fontFamily": "System",
                "fontSize": 17,
                "fontWeight": "700",
                "marginLeft": undefined,
                "paddingLeft": 4,
                "paddingTop": 1,
                "textAlign": "center",
              },
              undefined,
              undefined,
            ]
          }
        >
           
          Test
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "flex-end",
            "alignSelf": "center",
            "flex": 1,
            "flexDirection": "row",
            "justifyContent": "flex-end",
          }
        }
      />
    </View>
  </View>
</View>
`;
